src/Components/PaymentForm.jsx

```diff
@@ -0,0 +1,57 @@
+import React, { Component } from 'react'
+import { Elements, StripeProvider } from 'react-stripe-elements'
+import CheckoutForm from '../Components/CheckoutForm'
+import { Container, Button } from 'semantic-ui-react'
+import { connect } from 'react-redux';
+
+class PaymentForm extends Component {
+  state = {
+    renderPaymentForm: false
+  }
+
+  renderPaymentForm = () => {
+    this.setState({
+      renderPaymentForm: !this.state.renderPaymentForm
+    })
```

Comment:
This is a bad practice. React does setState asynchronously. It does when it has free resources. Pointing to `this.state` could lead to unintended end results. You should use prevState:

```
this.setState(prevState => ({renderPaymentForm: !prevState.renderPaymentForm )}) 
```